src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SymbolEquivalenceComparer.EquivalenceVisitor.cs

```diff
@@ -46,18 +46,14 @@ public bool AreEquivalent(ISymbol x, ISymbol y, Dictionary<INamedTypeSymbol, INa
                 }
 #endif
 
-                // https://github.com/dotnet/roslyn/issues/39643 This is a temporary workaround sufficient to get existing tests passing.
-                // This component should be modified to properly deal with differences caused by nullability.
-                if (x is ITypeSymbol xType && y is ITypeSymbol yType && xType.IsDefinition != yType.IsDefinition)
-                {
-                    if (x.IsDefinition)
-                    {
-                        y = yType.WithNullableAnnotationFrom(xType);
-                    }
-                    else
-                    {
-                        x = xType.WithNullableAnnotationFrom(yType);
-                    }
+                if (x is ITypeSymbol xType && y is ITypeSymbol yType)
+                {
+                    // Nullability is not considered a distinguishing factor between symbols in this component.  Strip
+                    // nullability from these symbols.  This also ensures we can do reference-equality checks later as
+                    // stripping nullability returns the underlying symbol and does not produce new symbols with the
+                    // updated nullability value.
+                    x = xType.WithNullableAnnotation(NullableAnnotation.None);
+                    y = yType.WithNullableAnnotation(NullableAnnotation.None);
                 }
```

Comment:
haven't been able to find a repro yet.  but this is the only change here in ages.  the new code is what we want here anyways.